bin/compile/function_metadata/metadata_builder.dart

```diff
@@ -45,22 +48,29 @@ Iterable<CompileTimeFunctionData> getFunctionData(
 
     for (final parameter in parameterList.parameters.skip(1)) {
       if (parameter.identifier == null) {
-        // Parameters must have a name to be used
+        // Parameters must have a name to be used. Skip this function.
         continue outerLoop;
       }
 
-      Iterable<Annotation> annotationsWithType(int type) => parameter.metadata.where((node) =>
-          (node.elementAnnotation?.element is ConstructorElement &&
-              getId((node.elementAnnotation!.element as ConstructorElement)
-                      .enclosingElement
-                      .thisType) ==
-                  type) ||
-          (node.elementAnnotation?.element is ConstVariableElement &&
-              getId((node.elementAnnotation!.element as ConstVariableElement)
-                      .evaluationResult!
-                      .value!
-                      .type) ==
-                  type));
+      /// Extracts all the annotations on a parameter that have a type with the type id [type].
+      Iterable<Annotation> annotationsWithType(int type) => parameter.metadata.where(
+            (node) =>
+                (node.elementAnnotation?.element is ConstructorElement &&
+                    getId(
+                          (node.elementAnnotation!.element as ConstructorElement)
+                              .enclosingElement
+                              .thisType,
+                        ) ==
+                        type) ||
+                (node.elementAnnotation?.element is ConstVariableElement &&
+                    getId(
+                          (node.elementAnnotation!.element as ConstVariableElement)
+                              .evaluationResult!
+                              .value!
+                              .type,
+                        ) ==
+                        type),
+          );
```

Comment:
can this be broken into separate parts because its unreadable and hard to follow in current form